pkg/asset/agent/oc.go

```diff
@@ -0,0 +1,50 @@
+package agent
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// ExecuteOC will execute an oc command.
+func ExecuteOC(pullSecret string, command []string) (string, error) {
+	// create registry config
+	ps, err := os.CreateTemp("", "registry-config")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		ps.Close()
+		os.Remove(ps.Name())
+	}()
+	_, err = ps.Write([]byte(pullSecret))
+	if err != nil {
+		return "", err
+	}
+	// flush the buffer to ensure the file can be read
+	ps.Close()
+	registryConfig := " --registry-config=" + ps.Name()
+	command = append(command, registryConfig)
+
+	var stdoutBytes, stderrBytes bytes.Buffer
+	// squash command into a string, then run in a shell. We do this to handle nested spaces in args
+	cmd := exec.Command("sh", "-c", strings.Join(command, " ")) //nolint:gosec
```

Comment:
gosec is right to complain here.
```suggestion
	cmd := exec.Command(command...)
```